system-test/storage.ts

```diff
@@ -780,6 +780,120 @@ describe('storage', () => {
     });
   });
 
+  describe('public access prevention', () => {
+    let bucket: Bucket;
+
+    const PUBLIC_ACCESS_PREVENTION_UNSPECIFIED = 'unspecified';
+    const PUBLIC_ACCESS_PREVENTION_ENFORCED = 'enforced';
+
+    const createBucket = () => {
+      bucket = storage.bucket(generateName());
+      return bucket.create();
+    };
+
+    const setPublicAccessPrevention = (bucket: Bucket, configuration: string) =>
+      bucket.setMetadata({
+        iamConfiguration: {
+          publicAccessPrevention: configuration,
+        },
+      });
+
+    const validateUnexpectedPublicAccessPreventionValueError = (
+      err: ApiError
+    ) => {
+      assert.strictEqual(err.code, 400);
+      return true;
+    };
+
+    const validateConfiguringPublicAccessWhenPAPEnforcedError = (
+      err: ApiError
+    ) => {
+      assert.strictEqual(err.code, 412);
+      return true;
+    };
+
+    before(createBucket);
+
+    it('inserts a bucket with enforced public access prevention', async () => {
```

Comment:
Looks like this test also checks that the bucket cannot be made public-- I'm thinking that should be included in the description here?